manifests/init.pp

```diff
@@ -2,8 +2,20 @@
 #
 class lvm (
   $volume_groups = {},
+  $version       = 'installed'
 ) {
 
+  $real_provider = $::osfamily ? {
+    'Debian' => 'apt',
+    'RedHat' => 'yum'
+  }
```

Comment:
This block should be removed, you shouldn't need to specify the provider, puppet defaults should be fine.